feat: implement HTTPClientWithHeaders method

hbagdi opened the pull request:
Multiple users of this library implement custom RoundTrippers to inject
authentication headers to authenticate against Kong's Admin API.

tjrivera commented on kong/utils.go:
```diff
@@ -51,3 +52,36 @@ func stringArrayToString(arr []*string) string {
+
+// RoundTripperWithHTTPHeaders returns a client which injects headers
+// before sending any request.
+func HTTPClientWitHeaders(client http.Client,
```
I think this might be a typo, do we want `HTTPClientWithHeaders`? Otherwise -- yes, this is useful!

tjrivera replied:
I also wonder if it might be useful to use the default http.Client if `client == nil` as well as a default transport.

Something like: 
```go
// HTTPClientWithHeaders returns a client which injects headers
// before sending any request.
func HTTPClientWithHeaders(client *http.Client,
	headers http.Header) http.Client {
	var res http.Client
	if client == nil {
		defaultTransport := http.DefaultTransport.(*http.Transport)
		res.Transport = defaultTransport
	} else {
		res = *client
	}
	res.Transport = headerRoundTripper{
		headers: headers,
		rt:      client.Transport,
	}
	return res
}
```

Otherwise, I think secure client initialization would look something like:

```go
	c := &http.Client{}
	defaultTransport := http.DefaultTransport.(*http.Transport)
	c.Transport = defaultTransport
	secureClient := HTTPClientWitHeaders(
		*c,
		http.Header{
			"kong-admin-token": []string{"my-token"},
		},
	)
	client, err := NewClient(nil, &secureClient)
```

hbagdi replied:
Excellent point. Thanks!
Updated the code and proceeding with merge now.

hbagdi committed:
feat: implement HTTPClientWithHeaders method

Multiple users of this library implement custom RoundTrippers to inject
authentication headers to authenticate against Kong's Admin API.
